### PR TITLE
Fix outdated Dockerfile and Flask app

### LINKED ISSUE #245: The included Dockerfile and associated Flask app is out of date in multiple ways.
I slogged through this and I have a working Dockerfile and a new Flask app to go with it.
I don't have much time to deal with making a pull request and it's not clear to me if the pull requests are being integrated.
I am watching this repository, if you want me to help you please comment and I'll try to be responsive.
If there's demand and a chance that the pull request will be integrated, I'll do my best to help.

### PR DESCRIPTION
**Pull request by Paulooh007**
## Fixes.
Fixes #245 and #240 

## Description
Here's what's included in this PR:
- An updated Dockerfile that uses the [`laser_encoders`](https://test.pypi.org/search/?q=laser_encoders) library.
- A working Flask app that provides a REST interface to the embed task.
- An updated README on how to use the laser docker image.

### CODE REVIEW

File: docker/Dockerfile
Comment (fissoreg) [inline docker/Dockerfile:3]: Why did you remove the `MAINTAINER` field? I don't know if it is actually needed, but I guess we need to make sure about this.

File: docker/Dockerfile
Comment (fissoreg) [inline docker/Dockerfile:8]: Choosing the Python version is not something to be done lightly. The issue references Python 3.7 and the repository also indicates that LASER is compatible with Python >= 3.7. So why 3.9 here? Are there any specific reasons for this?

**Review by fissoreg — COMMENTED** [review]
This looks very nice to me! :)
I left a couple of comments for which we probably need @heffernankevin and @avidale opinion.

File: docker/Dockerfile
Comment (Paulooh007) [reply docker/Dockerfile:3]: Oh, My apologies. It was a mistake while making multiple updates to the file.
I’ll add it now.

File: docker/Dockerfile
Comment (Paulooh007) [reply docker/Dockerfile:8]: Yes the issue referenced python 3.7, but the new `laser_encoders` library also indicates >=3.8 in the [readme](https://github.com/facebookresearch/LASER/blob/MLH-dev/laser_encoders/README.md). I used 3.9 because I thought it falls in range, no strong reason in particular. If Python 3.8 is preferred, I can switch.

**Review by avidale — CHANGES_REQUESTED** [review]
I have added a few comments that I would like to be addressed either in this request or in the follow-up requests. 

However, I believe that it would be better to freeze this one for now, and return to it after #249 is merged, because reusing a locally installed version of `laser_encoders` module would give us more freedom and consistency.

File: .github/workflows/lint_and_tests.yml
Comment (avidale) [inline .github/workflows/lint_and_tests.yml:1]: @Paulooh007 could you please merge the `main` branch into yours?
This way, we will get rid of all the "files changed" that don't have any relation to your changes in this PR.

File: docker/app.py
Comment (avidale) [inline docker/app.py:23]: In `laser_encoders`, we use either 3-letter codes (like `eng`) or 8-letter codes (like `eng_Latn`).
Adding 2-letter codes to this list is a good idea, but let's do that in another PR (here is an issue: https://github.com/facebookresearch/LASER/issues/273). 

And for this app right now, I would just explain that we need to use 3 or 8-letter codes instead.

File: docker/app.py
Comment (avidale) [reply docker/app.py:30]: Yes, exactly!

File: docker/app.py
Comment (avidale) [inline docker/app.py:38]: If I understand correctly, this variable equals the full code of the requested language, if it is in LASER2 and if this exact language has been already used, and None otherwise. And the cache used only if this variable is set. 

This seems to lead to 2 consequences:
* Models for LASER3 languages are never cached, but I would like them to. 
* The LASER2 model is not reused if we switch between different LASER2 languages, but I would like it to be reused.

If I understood your code incorrectly, please explain to me how it works.

**Review by avidale — APPROVED** [review]

**Pull request merged**
